MoreLinq.Test/ToLookupTest.cs

```diff
@@ -0,0 +1,118 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2008 Jonathan Skeet. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+using System;
+using System.Collections.Generic;
+using NUnit.Framework;
+
+namespace MoreLinq.Test
+{
+    [TestFixture]
+    public class ToLookupTest
+    {
+        [Test]
+        public void ToLookupWithNullKeyValuePairSequence()
+        {
+            Assert.ThrowsArgumentNullException("source", () =>
+                MoreEnumerable.ToLookup((IEnumerable<KeyValuePair<object, object>>) null));
```

Comment:
This tests an exception that is actually thrown by the real `ToLookup`, as the implementation doesn't check its arguments. I now see the same is true about the `ToDictionary` overload. Conceptually, I find it wrong to test behavior from the underlying library. What if in some version of the runtime the first argument to `ToLookup` is not named "source"?